verifier/src/report_body.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2018-2023 The MobileCoin Foundation
+
+//! Verifiers which operate on the [`ReportBody`](::mc_attest_core::ReportBody)
+//! data structure.
+
+use crate::{Result, VerificationStep};
+use mc_sgx_core_types::{Attributes, ReportBody};
+
+/// Verify the report body is as expected.
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ReportBodyVerifier<'a> {
+    expected_report_body: ReportBody,
+    report_body: &'a ReportBody,
+}
+
+impl<'a> ReportBodyVerifier<'a> {
+    /// Create a new [`ReportBodyVerifier`] instance.
+    ///
+    /// # Arguments:
+    /// * `expected_report_body` - The expected report body.
+    /// * `report_body` - The report body to verify conforms to the
+    ///   `expected_report_body`.
+    pub fn new(expected_report_body: ReportBody, report_body: &'a ReportBody) -> Self {
```

Comment:
I think we're still narrowing in on what the final API should be, for now this is using `ReportBody`, but we're moving toward a higher level `Evidence` or similar. I would like to flush out a few more of the lower level elements in the ReportBody and see how that flushes out.
The main reason being that there are multiple reports in the evidence and I'm not quite convinced  we want a `QEReportAttributesVerifier` and a `AppReportAttributesVerifier`

Reply:
There's two steps:

1. Extract data from the Evidence.
1. Check that evidence against expected data.

So my off-the-cuff thoughts is to add a new `Accessor` trait:

```rust
trait Accessor<T> {
    fn get(evidence: &Evidence) -> T;
}

struct QeReportAttributesAccessor;
impl Accessor<Attributes> for QeReportAttributesAccessor {
    fn get(evidence: &Evidence) -> Attributes {
        /* ... */
    }
}

struct AttributesVerifier<T: Accessor<Attributes>> {
    expected: Attributes,
    _marker: PhantomData<T>
}

impl<T: Accessor<Attributes>> Verifier for Attributes<T> {
    type Error = AttributesError;
    fn verify(&self, evidence: &Evidence) -> Result<(), AttributesError> {
        if (self.expected == T::get(evidence)) {
            Ok(())
        } else {
            Err(AttributesError)
        }
    }

type QeReportAttributesVerifier = AttributesVerifier<QeReportAttributesAccessor>;
type AppAttributesVerifier = AttributesVerifier<AppAttributesAccessor>;
```

Reply:
Added as a task to #7, unless there is strong preference for now, I would like to get one or 2 more sub pieces in and then see how it looks.